index.js

```diff
@@ -76,13 +82,76 @@ if (process.platform === 'win32') {
 
 getFn = function () {
 	var fn = function (/*…msg*/) {
-		var start = '', end = '';
-		forEach(fn._cliColorData, function (mod) {
-			end = '\x1b[' + mod[1] + 'm' + end;
-			start += '\x1b[' + mod[0] + 'm';
-		}, null, true);
-		return start + join.call(arguments, ' ') + end;
+		var start = ''
+		  , end   = ''
+		  , keys  = Object.keys(fn._cliColorData)
+		  , msg   = join.call(arguments, ' ');
+
+		if (keys.length) {
+			for (var i=0; i<keys.length; i++) {
+				start += ( i !== 0 ? ';' : '' ) + fn._cliColorData[keys[i]][0];
+				end   += ( i !== 0 ? ';' : '' ) + fn._cliColorData[keys[i]][1];
```

Comment:
I think we should stay away from micro-benchmarks, you can invalidate almost any code with such.

Do you have a case of real world code, that noticeably performs slower with code as written with `forEach`. I haven't got any reports about that, and in my opinion readability should not be affected for no reason.

Anyway my question here, was not about that, but about join of commands within one `\x1b[` separated with `;`.
I was not aware that we can freely do joins like that. Is this a common standard in ANSI?